agent/cvms/api/grpc/client.go

```diff
@@ -58,6 +98,43 @@ func (client *CVMSClient) Process(ctx context.Context, cancel context.CancelFunc
 	return eg.Wait()
 }
 
+func (client *CVMSClient) loadPendingMessages() error {
+	file := filepath.Join(client.storageDir, pendingMsgFile)
+	data, err := os.ReadFile(file)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(data, &client.pendingMsgs)
+}
+
+func (client *CVMSClient) savePendingMessages() error {
```

Comment:
Move storage to a separate package and add the interface so it can be mocked and tested more easily.